clients/admin-ui/src/features/dataset/DatasetCollectionView.tsx

```diff
@@ -0,0 +1,71 @@
+import { Box, Select, Spinner } from "@fidesui/react";
+import { ChangeEvent, useEffect, useState } from "react";
+
+import { FidesKey } from "../common/fides-types";
+import { useGetDatasetByKeyQuery } from "./dataset.slice";
+import DatasetFieldsTable from "./DatasetFieldsTable";
+import { DatasetCollection } from "./types";
+
+const useDataset = (key: FidesKey) => {
+  const { data, isLoading } = useGetDatasetByKeyQuery(key);
+
+  return {
+    isLoading,
+    dataset: data,
+  };
+};
+
+interface Props {
+  fidesKey: FidesKey;
+}
+
+const DatasetCollectionView = ({ fidesKey }: Props) => {
+  const { dataset, isLoading } = useDataset(fidesKey);
+  const [activeCollection, setActiveCollection] = useState<
+    DatasetCollection | undefined
```

Comment:
Should this be `DatasetCollection | []`?

Reply:
Since this is setting an `activeCollection`, it's just the one collection, so an object, not an array

Reply:
Yup, this is what the object's type looks like: https://github.com/ethyca/fides/blob/main/clients/admin-ui/src/features/dataset/types.ts/#L41-L48

I prefer `undefined` over `{}` since IMO it's clearer about the object not being defined yet. I find the typescript errors that result from that are clearer, as opposed to `{}` which typically end up with longwinded messages about "field1 is missing from... field2 is missing from..." Also, `{}` is truthy which means you can't just do `if(!datasetCollection)` which always trips me up